core/events.js

```diff
@@ -336,6 +336,40 @@ Blockly.Events.fromJson = function(json, workspace) {
  * @constructor
  */
 Blockly.Events.Abstract = function(elem) {
+  /**
+   * The block id for the block this event pertains to, if appropriate for the
+   * event type.
+   * @type {string|undefined}
+   */
+  this.blockId = undefined;
+
+  /**
+   * The variable id for the variable this event pertains to. Only set in
+   * VarCreate, VarDelete, and VarRename events.
+   * @type {string|undefined}
```

Comment:
@NeilFraser is there a shorthand for this when it's not a function argument?

Reply:
Fascinating.  I had assumed that {string=} would be equivalent, but it is not.  Closure uses {string|undefined} exclusively.